avocado/plugins/runner_nrunner.py

```diff
@@ -169,6 +171,25 @@ class Runner(RunnerInterface):
     name = 'nrunner'
     description = 'nrunner based implementation of job compliant runner'
 
+    @staticmethod
+    def _update_avocado_configuration_used_on_runnables(runnables, config):
+        """Checks if runnables have runner requirements fulfilled
+
+        :param runnables: the tasks whose runner requirements will be checked
+        :type runnables: list of :class:`Runnable`
+        :param config: A config dict to be used on the desired test suite.
+        :type config: dict
+        """
+        for runnable in runnables:
+            command = RUNNERS_REGISTRY_STANDALONE_EXECUTABLE.get(runnable.kind)
+            if command is None:
+                continue
+            command = " ".join(command)
+            configuration_used = STANDALONE_EXECUTABLE_CONFIG_USED.get(command)
+            for config_item in configuration_used:
+                if config_item in config:
```

Comment:
What happens when the `config_item` missing in `config` but it is necessary for runner?

Reply:
Well, can you think of when that would happen, other than when a coding mistake was made?  My point is:
* once a setting is created/registered, it will have a default value
* if a developer writes a runner expecting Avocado to forward a non-existing setting, it's on the developer to deal with the "undefined" behavior.

Reply:
Ok, I understand your logic, and I am ok with that. But then IMO this `if` statement is not needed, because when the `config_item` is not in `config` it has to be a coding mistake.